From 49250c8e0ef563aafd5bbea62612f3d15fdee3af Mon Sep 17 00:00:00 2001
From: Jeff Dike <jdike@x86_64.user-mode-linux.org>
Date: Wed, 14 Jul 2010 14:35:52 -0400
Subject: [PATCH] sysvinit - Remove sulogin dependency on /usr/lib*/libcrypt.a

Upstream-Status: Inappropriate [configuration]

# The src Makefile was checking for libcrypt.a on the host, not in the
# build environment.  This patch checks for $LCRYPT in the environment
# and uses it if it's there.
# - jdike@linux.intel.com
---
 src/Makefile | 12 ++----------
 1 file changed, 2 insertions(+), 10 deletions(-)

diff --git a/src/Makefile b/src/Makefile
index 0210171..f696247 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -101,16 +101,8 @@ endif
 
 ROOT	?= $(DESTDIR)
 
-# Additional libs for GNU libc.
-ifneq ($(wildcard $(ROOT)/usr/lib*/libcrypt.*),)
-  SULOGINLIBS	+= -lcrypt
-endif
-
-# Additional libs for GNU libc / multiarch on Debian based systems.
-ifneq ($(wildcard $(ROOT)/usr/lib/*/libcrypt.*),)
-ifneq ($(findstring -lcrypt, $(SULOGINLIBS)), -lcrypt)
-  SULOGINLIBS	+= -lcrypt
-endif
+ifneq ($(LCRYPT),)
+  SULOGINLIBS	+= $(LCRYPT)
 endif
 
 all:		$(BIN) $(SBIN) $(USRBIN)
-- 
2.25.1

